packages/snaps-jest/src/environment.ts

```diff
@@ -0,0 +1,161 @@
+
+/* eslint-disable */
+declare global {
+  const browser: WebdriverIO.Browser;
+  const snapsEnvironment: SnapsEnvironment;
+}
+/* eslint-enable */
+
+export class SnapsEnvironment extends NodeEnvironment {
+  // `browser` is always set in the environment setup function. To avoid needing
+  // to check for `undefined` everywhere, we use a type assertion here.
+  browser!: WebdriverIO.Browser;
+
+  #options: SnapsEnvironmentOptions;
+
+  #server: Server | undefined;
+
+  /**
+   * Constructor.
+   *
+   * @param options - The environment options.
+   * @param context - The environment context.
+   */
+  constructor(options: JestEnvironmentConfig, context: EnvironmentContext) {
+    super(options, context);
+    this.#options = getOptions(options.projectConfig.testEnvironmentOptions);
+  }
+
+  /**
+   * Set up the environment. This starts the built-in HTTP server, and creates a
+   * new browser instance.
+   */
+  async setup() {
+    await super.setup();
+
+    if (this.#options.server.enabled) {
+      this.#server = await startServer(this.#options.server);
+    }
+
+    const args = [];
+    if (this.#options.browser.headless) {
+      args.push('--headless');
+    }
+
+    this.browser = await remote({
+      logLevel: 'error',
+      capabilities: {
+        browserName: 'chrome',
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        'goog:chromeOptions': {
+          args,
+        },
+      },
+    });
+
+    this.global.browser = this.browser;
+    this.global.snapsEnvironment = this;
+  }
+
+  /**
+   * Tear down the environment. This closes the browser, and stops the built-in
+   * HTTP server.
+   */
+  async teardown() {
+    if (this.#options.keepAlive) {
+      return;
+    }
+
+    await this.browser?.deleteSession();
+    this.#server?.close();
+
+    await super.teardown();
+  }
+
+  /**
+   * Get the URL to the simulator, including the environment URL.
+   *
+   * @returns The simulator URL.
+   * @throws If the server is not running.
+   */
+  get url() {
+    assert(this.#server, 'Server is not running.');
+
+    const { port } = this.#server.address() as AddressInfo;
+    const simulatorUrl =
+      this.#options.simulatorUrl ?? `http://localhost:${port}/simulator/`;
+
+    const executionEnvironmentUrl =
+      this.#options.executionEnvironmentUrl ??
+      `http://localhost:${port}/environment/`;
+
+    return `${simulatorUrl}?environment=${encodeURIComponent(
+      executionEnvironmentUrl,
+    )}`;
+  }
+
+  /**
+   * Get the snap ID for the current environment. This assumes that the built-in
+   * server is running.
+   *
+   * @returns The snap ID.
+   * @throws If the server is not running.
+   */
+  get snapId() {
+    assert(
+      this.#server,
+      'You must specify a snap ID, because the built-in server is not running.',
+    );
+
+    const { port } = this.#server.address() as AddressInfo;
+    return `local:http://localhost:${port}`;
```

Comment:
Potentially we would want to support NPM snaps too, we don't need to handle it right now though

Reply:
It is, just not automatically. This snap ID is only used when no snap ID is passed to `installSnap`.

Reply:
Ah 🤔